staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -712,28 +810,44 @@ func (c *Cacher) startDispatching(event *watchCacheEvent) {
 	// gain from avoiding memory allocations is much bigger.
 	c.watchersBuffer = c.watchersBuffer[:0]
 
-	// Iterate over "allWatchers" no matter what the trigger function is.
-	for _, watcher := range c.watchers.allWatchers {
-		c.watchersBuffer = append(c.watchersBuffer, watcher)
-	}
-	if supported {
-		// Iterate over watchers interested in the given values of the trigger.
-		for _, triggerValue := range triggerValues {
-			for _, watcher := range c.watchers.valueWatchers[triggerValue] {
-				c.watchersBuffer = append(c.watchersBuffer, watcher)
+	if event.Type == watch.Bookmark {
```

Comment:
nit: adding this branch makes the indentation looks a little bit crazy. ideally, we probably should add a two helper funcs or try to reduce the indentations for readability.

Reply:
add a helper function here.